server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/service/DeleteMailsFromMailQueueTask.java

```diff
@@ -0,0 +1,135 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.queue.api.MailQueue;
+import org.apache.james.queue.api.ManageableMailQueue;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Booleans;
+
+public class DeleteMailsFromMailQueueTask implements Task {
+
+    public static class AdditionalInformation implements TaskExecutionDetails.AdditionalInformation {
+        private final String mailQueueName;
+        private final Supplier<Long> countSupplier;
+        private final long initialCount;
+
+        private final String sender;
+        private final String name;
+        private final String recipient;
+
+        public AdditionalInformation(String mailQueueName, Supplier<Long> countSupplier,
+                                     Optional<MailAddress> maybeSender, Optional<String> maybeName,
+                                     Optional<MailAddress> maybeRecipient) {
+            this.mailQueueName = mailQueueName;
+            this.initialCount = countSupplier.get();
+            this.countSupplier = countSupplier;
+
+            sender = maybeSender.map(MailAddress::asString).orElse(null);
```

Comment:
Why would we want to use null for these fields ?

Reply:
I declared them as final fields, then they have to be initiated in Constructors. Is there any problem with a null field?

Reply:
Yes. We should be able to format Optionals into JSON